src/parser/jobs/blu/modules/AlwaysBeCasting.tsx

```diff
@@ -105,74 +107,60 @@ export class AlwaysBeCasting extends CoreAlwaysBeCasting {
 		// the full 5000ms channel, BUT, if they used the kick, then they should
 		// have waited at least 4000ms for most of the channel to have happened.
 
-		const currentFlurry = this.currentPhantomFlurry ?? this.phantomFlurryHistory[this.phantomFlurryHistory.length - 1]
-		if (currentFlurry === null) { return }
-		currentFlurry.manualKick = true
+		const allChannels = this.channelHistory.entries
+		if (allChannels.length === 0) { return }
+		const currentChannel = allChannels[allChannels.length - 1]
+		if (currentChannel === null) { return }
+		currentChannel.data.manualKick = true
 	}
 
-	private onApplyPhantomFlurry(event: Events['action']) {
-		if (this.currentPhantomFlurry != null) { return }
-
-		this.currentPhantomFlurry = {
-			start: event.timestamp,
-			end: event.timestamp + this.data.statuses.PHANTOM_FLURRY.duration,
-			manualKick: false,
-			inMoonFlute: this.actors.current.hasStatus(this.data.statuses.WAXING_NOCTURNE.id),
-		}
-		const anyActionFilter = filter<Event>()
-			.source(this.parser.actor.id)
-			.action(noneOf([this.data.actions.PHANTOM_FLURRY.id]))
-			.type('action')
-		this.phantomFlurryInterruptingActionHook = this.addEventHook(anyActionFilter, this.onRemovePhantomFlurry)
-
-		this.phantomFlurryHistory.push(this.currentPhantomFlurry)
+	private onApplyChannel(event: Events['action']) {
+		const newChannel = this.channelHistory.openNew(event.timestamp)
+		newChannel.data.inMoonFlute = this.actors.current.hasStatus(this.data.statuses.WAXING_NOCTURNE.id)
+		newChannel.data.actionId = event.action
 	}
 
-	private onRemovePhantomFlurry(event: Events['statusRemove'] | Events['action']) {
-		if (this.currentPhantomFlurry == null) {
-			return
-		}
-		if (this.phantomFlurryInterruptingActionHook == null) {
-			return
-		}
-
-		this.currentPhantomFlurry.end = event.timestamp
-		this.removeEventHook(this.phantomFlurryInterruptingActionHook)
-
-		this.currentPhantomFlurry = undefined
-		this.phantomFlurryInterruptingActionHook = undefined
-
+	private onRemoveChannel(event: Events['statusRemove']) {
+		this.channelHistory.closeCurrent(event.timestamp)
 	}
 
 	override considerCast(action: Action, castStart: number): boolean {
 		if (action === this.data.actions.PHANTOM_FLURRY) {
 			this.debug(`Phantom Flurry began channeling at ${this.parser.formatEpochTimestamp(castStart)}`)
 			return false
 		}
+		if (action === this.data.actions.APOKALYPSIS) {
+			this.debug(`Apokalypsis began channeling at ${this.parser.formatEpochTimestamp(castStart)}`)
```

Comment:
If you do the `channelCastIds` suggestion, you can merge this condistion with the other as `this.channelCastIds.includes(action.id)` then just do `action.name + ' began channeling at`... for your message, rather than needing to duplicate the code for each channel we eventually have.

Reply:
Didn't go with the list idea per above, but did some changes so this won't be duplicated code anymore.

Reply:
Works for me for now. If/when we get a third we can revisit the list definition idea.